### PR TITLE
Pair programming - Task 2

### PR DESCRIPTION
**Pull request by AnsarIbrahim**
### This PR Contains :
- Test
- Add Test: 
   - Edit function tests.
   - Delete task function test.
   - ClearAll completed function tests.
- Add Jest Frameworks
- Linters Check
- Mock
- Used JSDOM
- Pair Programming
##### To get a local copy up and running:
1. git clone the repo.
2. Checkout to the correct branch (Test-2-Todolist).
3.  Run npm install.
4.  Run the npm test

### CODE REVIEW

File: src/tests/edit.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import edit from '../edit.js';
+import { saveLocal } from '../live.js';
+
+jest.mock('../live.js', () => ({
```
Comment (julie-ify) [inline src/tests/edit.test.js:4]: - [ ] You have done well adding tests for editing descriptions and removing all completed tasks, but you are missing tests for updating the `completed` status of a task. Kindly add that.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @AnsarIbrahim and @Bennyjoez ,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi team😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on completing this milestone and implementing what was requested by the previous reviewer

Cheers and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bennyjoez** [comment]
Thank you @MahmoudBakr23 @julie-ify for your time and help. We have learned because of it. Happy coding!! 🧑‍💻

**Pull request merged**